Functions/Administration/xASL_adm_CleanUpBeforeRerun.m

```diff
@@ -324,10 +324,10 @@ function xASL_adm_CleanUpBeforeRerun(AnalysisDir, iModule, bRemoveWMH, bAllSubje
 
         RemoveFields = {};
         if ~isempty(find(iModule==1)) % if we remove the structural provenance
-            RemoveFields{end+1} = 'Structural';
+            RemoveFields{1} = 'Structural';
```

Comment:
In case iModule is [ 2], then RemoveFields == {'ASL'}.

But then later in the code, you use iModule to index RemoveFields, so you call RemoveFields{2}, which obviously isn't 'ASL' but returns an error. The correct solution that solves everything is to create RemoveFields with the iModule indices, instead as a growing list.
That means - revert back to the change I made. You can of course keep RemoveFields = cell(0) - that is fine.

Reply:
Sorry, it wasn't an error. But it was an important warning.

Running cleanup on ASL only (that is [2] and not [1 2]) creates this warning and then skips a large part of this cleaning module.

`Cleaning up x-struct:100%Warning: Index exceeds the number of array elements. Index must not exceed 1. 
> In xASL_adm_CleanUpBeforeRerun (line 447)
In xASL_module_ASL (line 102)
In xASL_init_Iteration>runIteration (line 409)
In xASL_init_Iteration (line 146)
In ExploreASL_Process (line 138)
In ExploreASL (line 97) `

Either way - my fix fixes this.

Reply:
With your fix, if you only clean ASL, you get RemoveFields{EMPTY 'ASL'} which is not legal right?
The only legal options should be:

RemoveFields{'Structural'}
RemoveFields{'ASL'}
RemoveFields{'Structural' 'ASL'}

Reply:
You are indeed missing the point.

You have iModule == [2] on the input, then you create RemoveFields{'ASL'} and then you try to index RemoveFields{'ASL'} with iModule and RemoveFields{2} throws an error...

While my solution RemoveFields{empty 'ASL'} is indexed only with 2, so the empty is effectively unused.

The problem is that this doesn't really have another solution than I propose because you need to use this iModule as index and by creating the incremental list, you loose any info about order. 

Just test this and you'll see it for yourself. I'm of course fine with any other more complicated solution. But the current version doesn't work.

Reply:
Should be fixed now in 6c7dd738416cb5a2eb60df9db5a48a3073920c1a

Reply:
Not fixed. it's even worse. Please read my comment. You can't create a list RemoveFields incrementally and then index this with iModule that has completely different indices.